fairscale/nn/model_parallel/initialize.py

```diff
@@ -28,6 +28,8 @@
 
 from .utils import ensure_divisibility
 
+# FIXME: this assumes exactly one distributed process.
```

Comment:
I suggest reading the megatron paper first if you haven't already. I do think it is an approach that does model parallelism in a way that's layer-specific, not a general approach like ZERO 1/2/3.

Reply:
i.e. for different layers, vocab, embedding, etc., different model parallel versions will need to be written specifically.